File: skyspacefiftyfive_ui/src/services/LoungeAccessService.js
import http from "../http-common"; 

class LoungeAccessService {
  getAllLoungeAccesss(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/loungeAccess/loungeAccesss`, searchDTO);
  }

  get(loungeAccessId) {
    return this.getRequest(`/loungeAccess/${loungeAccessId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/loungeAccess?field=${matchData}`, null);
  }

  addLoungeAccess(data) {
    return http.post("/loungeAccess/addLoungeAccess", data);
  }

  update(data) {
  	return http.post("/loungeAccess/updateLoungeAccess", data);
  }
  
  uploadImage(data,loungeAccessId) {
  	return http.postForm("/loungeAccess/uploadImage/"+loungeAccessId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new LoungeAccessService();

File: src/main/java/com/skyspacefiftyfive/service/LoungeAccessService.java
package com.skyspacefiftyfive.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.skyspacefiftyfive.domain.LoungeAccess;
import com.skyspacefiftyfive.dto.LoungeAccessDTO;
import com.skyspacefiftyfive.dto.LoungeAccessSearchDTO;
import com.skyspacefiftyfive.dto.LoungeAccessPageDTO;
import com.skyspacefiftyfive.dto.LoungeAccessConvertCriteriaDTO;
import com.skyspacefiftyfive.service.GenericService;
import com.skyspacefiftyfive.dto.common.RequestDTO;
import com.skyspacefiftyfive.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface LoungeAccessService extends GenericService<LoungeAccess, Integer> {

	List<LoungeAccess> findAll();

	ResultDTO addLoungeAccess(LoungeAccessDTO loungeAccessDTO, RequestDTO requestDTO);

	ResultDTO updateLoungeAccess(LoungeAccessDTO loungeAccessDTO, RequestDTO requestDTO);

    Page<LoungeAccess> getAllLoungeAccesss(Pageable pageable);

    Page<LoungeAccess> getAllLoungeAccesss(Specification<LoungeAccess> spec, Pageable pageable);

	ResponseEntity<LoungeAccessPageDTO> getLoungeAccesss(LoungeAccessSearchDTO loungeAccessSearchDTO);
	
	List<LoungeAccessDTO> convertLoungeAccesssToLoungeAccessDTOs(List<LoungeAccess> loungeAccesss, LoungeAccessConvertCriteriaDTO convertCriteria);

	LoungeAccessDTO getLoungeAccessDTOById(Integer loungeAccessId);







}





